packages/core/src/extension/extension-action-creator.ts

```diff
@@ -0,0 +1,43 @@
+import { createAction, createErrorAction, ThunkAction } from '@bigcommerce/data-store';
+import { Observable, Observer } from 'rxjs';
+
+import { InternalCheckoutSelectors } from '../checkout';
+import { RequestOptions } from '../common/http-request';
+
+import { ExtensionAction, ExtensionActionType } from './extension-actions';
+import { ExtensionRequestSender } from './extension-request-sender';
+
+export class ExtensionActionCreator {
+    constructor(private _requestSender: ExtensionRequestSender) {}
+
+    loadExtensions(
+        options?: RequestOptions,
+    ): ThunkAction<ExtensionAction, InternalCheckoutSelectors> {
+        return (store) =>
+            Observable.create((observer: Observer<ExtensionAction>) => {
+                const state = store.getState();
+                const cart = state.cart.getCartOrThrow();
```

Comment:
Do we need to pass cartId for extensions? Aren't they something that belong at store level?

Reply:
I put it here to match with loading cart and payment methods.
Removed.
On another note, when we render an extension, we'll pass `cartId` and `extensionId` to an iframe as params.

Reply:
Yes that can we done at a different layer, if we don't need information as part of action creation we should avoid it.